programs/ziti-edge-tunnel/instance.c

```diff
@@ -306,6 +313,8 @@ static void setTunnelServiceAddress(tunnel_service *tnl_svc, ziti_service *servi
         for(int port_idx = 0; cfg_v1.port_ranges[port_idx]; port_idx++) {
             tnl_port_range_arr[port_idx] = getTunnelPortRange(cfg_v1.port_ranges[port_idx]);
         }
+        if (cfg_v1.addresses) free(cfg_v1.addresses);
```

Comment:
Is it necessary to free `addresses` (and `port_ranges`) here? What about the other list fields, like `protocols`? I'm guessing you want `free_ziti_intercept_cfg_v1(&cfg_v1)` here, so that all of the fields are freed.